src/v2i-hub/RSUHealthMonitorPlugin/src/RSUHealthMonitorWorker.cpp

```diff
@@ -0,0 +1,175 @@
+            RSUFieldOIDStruct rsuIFMStatus = {"rsuIFMStatus", RSU_IFM_STATUS, false};
+            rsuStatusTbl.push_back(rsuIFMStatus);
+
+            RSUFieldOIDStruct rsuMode = {"rsuMode", RSU_MODE, true};
+            rsuStatusTbl.push_back(rsuMode);
+
+            RSUFieldOIDStruct rsuChanStatus = {"rsuChanStatus", RSU_CHAN_STATUS, true};
+            rsuStatusTbl.push_back(rsuChanStatus);
+        }
+        return rsuStatusTbl;
+    }
+
+    bool RSUHealthMonitorWorker::isAllRequiredFieldsPresent(const RSUMibVersion &mibVersion, const vector<string> &fields) const
+    {
+        bool isAllPresent = true;
+        for (const auto &config : GetRSUStatusConfig(mibVersion))
+        {
+            if (config.required && find(fields.begin(), fields.end(), config.field) == fields.end())
+            {
+                isAllPresent = false;
+                PLOG(logWARNING) << "No broadcast as required field " << config.field << " is not present!";
+            }
+        }
+        return isAllPresent;
+    }
+
+    RSUStatusConfigTable RSUHealthMonitorWorker::GetRSUStatusConfig(const RSUMibVersion &mibVersion) const
+    {
+        RSUStatusConfigTable result;
+        try
+        {
+            result = _RSUSTATUSConfigMapPtr->at(mibVersion);
+        }
+        catch (const out_of_range &ex)
+        {
+            PLOG(logERROR) << "Unknown MIB version! " << ex.what();
+        }
+        return result;
+    }
+
+    std::map<double, double> RSUHealthMonitorWorker::ParseRSUGPS(const std::string &gps_nmea_data) const
+    {
+        std::map<double, double> result;
+        nmea::NMEAParser parser;
+        nmea::GPSService gps(parser);
+        try
+        {
+            parser.readLine(gps_nmea_data);
+            std::stringstream ss;
+            ss << std::setprecision(8) << std::fixed << gps.fix.latitude << std::endl;
+            auto latitude_str = ss.str();
+            std::stringstream sss;
+            sss << std::setprecision(8) << std::fixed << gps.fix.longitude << std::endl;
+            auto longitude_str = sss.str();
+            result.insert({std::stod(latitude_str), std::stod(longitude_str)});
+            PLOG(logDEBUG) << "Parse GPS NMEA string: " << gps_nmea_data << ". Result (Latitude, Longitude): (" << latitude_str << "," << longitude_str << ")";
+        }
+        catch (nmea::NMEAParseError &e)
+        {
+            fprintf(stderr, "Error:%s\n", e.message.c_str());
+        }
+        return result;
+    }
+
+    Json::Value RSUHealthMonitorWorker::getRSUStatus(const RSUMibVersion &mibVersion, const string &_rsuIp, uint16_t &_snmpPort, const string &_securityUser, const string &_authPassPhrase, const string &_securityLevel, long timeout)
+    {
+        auto rsuStatusConfigTbl = GetRSUStatusConfig(mibVersion);
+        if (rsuStatusConfigTbl.size() == 0)
+        {
+            PLOG(logERROR) << "RSU status update call failed due to the RSU status config table is empty!";
+            return Json::nullValue;
```

Comment:
Why are we printing error and returning null instead of throwing an exception

Reply:
We encourage users to update the RSU config table by updating the mib version with the supported version, but it is not a fatal program issue.

Reply:
Exceptions do not need to be fatal but this is fine for now.